tests/test_datacube.py

```diff
@@ -3,10 +3,26 @@
 sys.path.append('../scripts/')
 from datacube import *
 
-def test_read_data(data_path):
+@pytest.mark.parametrize('data_path,expected_mv', [
+    ('./data/good/wind/*.nc', False),
+    ('./data/good/pr/*.nc', False),
+    ('./data/bad/example1/*.nc', True),
+    ('./data/bad/example2/*.nc', True),
+    ('./data/bad/example3/*.nc', True)])
+@pytest.mark.xfail(raises=ValueError)
+def test_path(data_path, expected_mv):
+    missing_value = False
+
     ds = read_data(data_path)
 
-    for var in ['dv', 'rlat', 'rlon', 'lat', 'lon']: 
-        assert var in ds.variables, 'Key \'{}\' not properly inserted into Dataset'.format(var)
+    for var in ['dv', 'rlat', 'rlon', 'lat', 'lon']:
+        if var not in ds.variables:
+            missing_value = True
+    assert(missing_value == expected_mv), 'A key was not properly inserted into the Dataset'
+    assert(len(ds['dv'].shape) == 3), 'Incorrect dimension size after loading'
```

Comment:
This second assertion is actually a separate test and should be in a separate function.

Reply:
Yup I realized that. Coming soon to a commit near you...